compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ExpectedTypeFinder.java

```diff
@@ -480,11 +489,13 @@ public Optional<TypeSymbol> transform(ImplicitNewExpressionNode node) {
             return Optional.empty();
         }
 
+        // TODO Fix casting
```

Comment:
Should we create an issue for this todo?

Reply:
Addressed with 0297d2f1caaeea94fa6c0bc2af3c308d99f970b4